.github/workflows/CI.yml

```diff
@@ -1,6 +1,11 @@
-name: Test Action
+name: CI
 
-on: pull_request
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - feature/**
 
 env:
   RESOURCE_GROUP_NAME: "Prod-RefArch-RGOnly-TD"
```

Comment:
The suffix I am using for the RG name is "-<env><deployment type>"

For this suffix, I suggest <deployment type" should be a value that indicates GitHub Axctions (e.g. "GA", or just "G")

For <env> I have used "T" (for Test) and "P" (for Prod) to indicate which CM infra is creating the RG

For <deployment type> I have used "D" (for Direct Deployer) and "A" for Azure DevOps.

The purpose of the suffix was to allow the _same_ BP to be deployed concurrently via different deployment mechanisms. The BP may also have to handle naming certain resources with a globally unique name for each deployment. There are already example BPs doing this, or I can provide an overview.

After the deployment has finished I do not expect to see any RGs with a suffix ending "A" or "GA" (or whatever you choose) as these should be automatically destroyed. Only RGs with suffix ending "D" should exist after a successful test.

Happy to discuss if you have any concerns.

Reply:
Great thank you. We already have automated teardown of GA created resources so I'll fix the name (going to go with `G`) and then merge.